### ISSUE TITLE
Proper Active Directory - Password support

### ISSUE THREAD

**Issue by skol99** [Microsoft/azuredatastudio#2814]
Mac - Azure Data Studio Version: 1.0.0

Steps to Reproduce:

When connecting to a dbase, there is no Active Directory - Password option.   I have used the SQL 2019 preview to utilize the Azure Explorer, and it shows up all my databases, but as soon as I want to connect to them I'm prompted to enter a SQL Login.   My Azure AD Account already has permissions to connect to the databases on the server.   It has been set up such that each Azure AD user account is in one of 2 SQL groups ReadOnly and ReadWrite, the users then passthrough connectivity to SQL without the user needing to enter any further login.

Please let me know when this will be possible on Azure Data Studio.

Thanks

Simon

**Comment by MattIrv** [comment]
We're hoping to have AAD auth added soon, see https://github.com/Microsoft/azuredatastudio/issues/13 for more info. I'll close this issue as a duplicate of that one. Thanks!

**Comment by pleppert** [comment]
The linked issue is now closed, but in that build I still don't see Azure Active Directory - Password, only Azure Active Directory - Universal with MFA support. Am I missing a way to log in with that authentication type when otherwise my login requires a password? Or should this issue be re-opened since the issue it was closed as a duplicate of did not add the requested feature?

**Comment by MattIrv** [comment]
Are you able to login using the same username and password if you choose the existing "Azure Active Directory - Universal with MFA support" and then add the account?

**Comment by pleppert** [comment]
There is no option that I can see to give it my password. I had previously added my account through the Azure panel, but I have tried doing it through the "Add an account" option in the Account drop down on the Connection pane after selecting Azure Active Directory - Universal with MFA support. Both times I was directed to the microsoft site to enter a device ID and grant access to Azure Data Studio, which I did.

I have selected the database to connect to, so I am not trying to connect to master (although I get the same result there).

The error I actually receive is:

```
Login failed for user 'NT AUTHORITY\ANONYMOUS LOGON'.
```

When I connect through SSMS using the Azure Active Directory - Password option, I am given a password field to fill out which I do not see here.

**Comment by pleppert** [comment]
Thank you for the help.

It fails in SSMS with (mostly) the same message, "Login failed for user 'NT AUTHORITY\ANONYMOUS LOGON'. (Microsoft SQL Server, Error: 18456)"

Through SSMS I was not directed to a website, but a window popped up with an embedded browser control (apparently) to log in. It accepted my credentials there as correct, but then gave me this message. I did change the database to connect to (I would expect this error when trying to connect to master since my user does not have permissions) to a database that I do have permissions to when using the password authentication in SSMS.

**Comment by kburtram** [comment]
@aaomidi do recall the detail on why we couldn't support AAD - Password?  I think it was that this isn't supported by .Net Core SqlClient.  Is that correct?

**Comment by aaomidi** [comment]
@kburtram that is correct. https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/482

**Comment by SilentCreep** [comment]
@kburtram @MattIrv @aaomidi Is there any chance that AAD - Password will be supprted by ADS. As listed on that site 
https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication?view=sql-server-ver15#setting-azure-active-directory-authentication

To be more exact:

Value | Description | Framework | Microsoft.Data.SqlClient version
-- | -- | -- | --
Active Directory Password | Authenticate with an Azure AD identity by using a username and password | .NET Framework 4.6+, .NET Core 2.1+, .NET Standard 2.0+ | 1.0+

Thx in advance

**Comment by mz-bayer** [comment]
Issue #13 seems to be locked, so I'll add a use case to this report.

My customer has blocked the use of **AAD with MFA** and I can only use ADS with database user and password. Which in essence means that I cannot use it at all once the audition process starts on the system I am building.

Thanks,

Martin

**Comment by Zerglrisk** [comment]
When 'Azure Directory - Password' Support in AzureDataStudio. or nothing plan to support?? let me know

**Comment by aafvstam** [comment]
How is this still a 'new' issue from 2018. It is blocking us from moving from MS SSMS as it won't allow to connect Azure AD identity by using a username and password.

**Comment by ibakirov** [comment]
I would not expect new features for Azure Data Studio due to Azure Data Studio set to retirement: https://learn.microsoft.com/en-us/azure-data-studio/whats-happening-azure-data-studio
